### PR TITLE
Understand that a tuple is also a Container. With test.

### LINKED ISSUE #671: Make tuple types subtypes of Sized, Container, Reversible, etc.
Currently `(1, 1)` is compatible with `Sequence[int]` and `Iterable[int]` but not with `Sized` or `Container[int]`, which is inconsistent.

This code should be fine, but currently mypy complains about it:

```
from typing import Sized, Container
def f(x: Sized) -> None: pass
f((1, 2)) # Argument 1 to "f" has incompatible type "Tuple[int, int]"; expected "Sized"
def g(x: Container[int]) -> None: pass
g((1, 2)) # Argument 1 to "g" has incompatible type "Tuple[int, int]"; expected Container[int]
```

Once this has been implemented, also simplify the signature of `len()` to not have `tuple` explicitly (`Sized` should be enough for the argument).

### PR DESCRIPTION
**Pull request by gvanrossum**
Maybe this closes #671 ?

### CODE REVIEW

File: mypy/subtypes.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def visit_tuple_type(self, left: TupleType) -> bool:
                 return all(is_subtype(item, target_item_type)
                            for item in left.items)
             elif (is_named_instance(right, 'typing.Iterable') or
+                  is_named_instance(right, 'typing.Sized') or
```
Comment (JukkaL) [inline mypy/subtypes.py]: This isn't right as `Sized` isn't generic, unlike the other types, so `right.args[0]` will fail below. May be worth adding a test case for this.

Comment (JukkaL) [inline mypy/subtypes.py]: There is a test case already. Hmm...

Comment (gvanrossum) [inline mypy/subtypes.py]: Hm... Weird. If I make Sized non-Generic I indeed get an internal error.
But if I remove the check for typing.Sized from subtypes.py, the test I
wrote fails, with
```
error: Incompatible types in assignment (expression has type "Tuple[]",
variable has type "Sized")
```
I think I have a fix, wait a sec.

Comment (gvanrossum) [inline mypy/subtypes.py]: Try the 3rd commit.

On Mon, Dec 14, 2015 at 3:06 PM, Guido van Rossum <guido@python.org> wrote:

> Hm... Weird. If I make Sized non-Generic I indeed get an internal error.
> But if I remove the check for typing.Sized from subtypes.py, the test I
> wrote fails, with
> ```
> error: Incompatible types in assignment (expression has type "Tuple[]",
> variable has type "Sized")
> ```
> I think I have a fix, wait a sec.
>



-- 
--Guido van Rossum (python.org/~guido)

**Comment by JukkaL** [comment]
Are there merge conflicts? It seems like the travis build didn't run.

**Comment by gvanrossum** [comment]
Eh, I always do this. I reuse the forked repo but I forget to update it to the latest master before pushing a new branch. Merged now, and tests are now running.

**Comment by gvanrossum** [comment]
Ooh, now the tests fail because I added bool to lib-stub/builtins.py. I'll
see what I can do about this.

**Comment by gvanrossum** [comment]
OK, another try...

On Mon, Dec 14, 2015 at 5:05 PM, Guido van Rossum <guido@python.org> wrote:

> Ooh, now the tests fail because I added bool to lib-stub/builtins.py. I'll
> see what I can do about this.
>



-- 
--Guido van Rossum (python.org/~guido)

**Comment by gvanrossum** [comment]
At last! Green!

**Pull request merged**